src/Backend/Modules/MediaLibrary/Domain/MediaItem/MediaItemSelectionDataGrid.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+namespace Backend\Modules\MediaLibrary\Domain\MediaItem;
+
+use Backend\Core\Engine\DataGridDatabase;
+use Backend\Core\Engine\DataGridFunctions as BackendDataGridFunctions;
+use Backend\Core\Engine\Model;
+use Backend\Core\Language\Language;
+use SpoonFormDropdown;
+
+/**
+ * @TODO replace with a doctrine implementation of the data grid
+ */
+class MediaItemSelectionDataGrid extends DataGridDatabase
+{
+    public function __construct(Type $type, int $folderId = null)
+    {
+        parent::__construct(
+            'SELECT i.id, i.storageType, i.type, i.url, i.title, i.shardingFolderName,
+                COUNT(gi.mediaItemId) AS num_connected, i.mime, UNIX_TIMESTAMP(i.createdOn) AS createdOn
+             FROM MediaItem AS i
+             LEFT OUTER JOIN MediaGroupMediaItem AS gi ON gi.mediaItemId = i.id
+             WHERE i.type = ?' . $this->getWhere($folderId) . ' GROUP BY i.id',
+            $this->getParameters($type, $folderId)
+        );
+
+        // filter on folder?
+        if ($folderId !== null) {
+            $this->setURL('&folder=' . $folderId, true);
+        }
+
+        $this->setExtras($type);
+    }
+
+    private function getColumnHeaderLabels(Type $type): array
```

Comment:
how much of this code is a duplicate of already existing code? Wouldn't it be better to extend that class and make them protected?

Reply:
I have created a branch where I extend the MediaItemDataGrid. I don't find it a very elegant solution but it can solve the problem of having 2 fully separate implementations of the data grid.

https://github.com/ohvitorino/forkcms/commit/746c144bc7c471a65844a5e0d8097917ff636787